### PR TITLE
feat: start simulation from plugin, closes MISC-301 and MISC-302

### PR DESCRIPTION
**Pull request by tpetillot**
chore: split up gatling plugin settings 
---

**Motivation:**
Too much configuration inside `GatlingPlugin`

**Modification:**
* Add settings package with all previous settings split up
* Reduce comments sizes

feat: start simulation from plugin 
---

**Modifications:**
* Upgrade plugin
* Update enterpriseUrl to root URL instead of public API endpoint
* Add start simulation task and related settings (simulation ID and properties)
* When simulation ID not configured, call simulation creation

**Ref:** MISC-301, MISC-302

### CODE REVIEW

**Comment by tpetillot** [comment]
⚠️ CI cannot pass until [gatling-enterprise-plugin-commons milestone release](https://github.com/gatling/gatling-enterprise-plugin-commons/runs/4522444846?check_suite_focus=true)

File: src/main/scala/io/gatling/sbt/settings/gatling/EnterpriseSettings.scala [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  private def httpEnterpriseClient(config: Configuration) = Def.task {
+    val settingUrl = new URL((config / enterpriseUrl).value.toExternalForm + PublicApiPath)
+    val settingApiToken = (config / enterpriseApiToken).value
+
+    if (settingApiToken.isEmpty) {
+      throw new IllegalStateException("Gatling / apiToken has not been specified")
+    }
+
+    val client = new OkHttpEnterpriseClient(settingUrl, settingApiToken)
+    client.checkVersionSupport(BuildInfo.version, BuildInfo.name)
```
Comment (ccedric) [inline src/main/scala/io/gatling/sbt/settings/gatling/EnterpriseSettings.scala:88]: it should be checkVersionSupport(name, version)

Comment (Isammoc) [reply src/main/scala/io/gatling/sbt/settings/gatling/EnterpriseSettings.scala:88]: named parameters?

**Review by ccedric — APPROVED** [review]

File: src/main/scala/io/gatling/sbt/settings/gatling/EnterpriseSettings.scala [hunk 2]
```diff
@@ -114,32 +123,36 @@ object EnterpriseSettings {
...
-         |enterprisePackageId := ${simulation.pkgId}
-         |
-         |Setting for enterpriseStart task next calls:
-         |enterpriseSimulationId := ${simulation.id}
-         |
-         |Successfully start simulation named '${simulation.name}' run.
-         |Live reports at ${baseUrl.toExternalForm + simulationAndRunSummary.runSummary.reportsPath}
-         |""".stripMargin
-    )
+    streams.value.log.success(s"Simulation successfully started; once running, report will be available at ${baseUrl.toExternalForm + simulationAndRunSummary.runSummary.reportsPath}")
```
Comment (ccedric) [inline src/main/scala/io/gatling/sbt/settings/gatling/EnterpriseSettings.scala:155]: report**s**

File: build.sbt [hunk 3]
```diff
@@ -9,7 +8,7 @@ githubPath := "gatling/gatling-sbt-plugin"
 libraryDependencies ++= Seq(
   "org.scalatest"     %% "scalatest"                         % "3.2.10" % Test,
   "org.zeroturnaround" % "zt-zip"                            % "1.14",
-  "io.gatling"         % "gatling-enterprise-plugin-commons" % "0.0.3"
+  "io.gatling"         % "gatling-enterprise-plugin-commons" % "0.0.3-M20211216111418"
```
Comment (slandelle) [inline build.sbt:11]: Please upgrade to 0.0.3-M20211217085936

**Review by slandelle — CHANGES_REQUESTED** [review]

**Comment by slandelle** [comment]
formatting issue

File: src/main/scala/io/gatling/sbt/GatlingKeys.scala [hunk 4]
```diff
@@ -37,6 +37,7 @@ object GatlingKeys {
   val enterpriseApiToken = settingKey[String]("API Token for package upload on Gatling Enterprise")
   val enterprisePackageId = settingKey[String]("Target package ID on Gatling Enterprise")
   val enterpriseDefaultSimulationClassname = settingKey[String]("Target simulation class name on Gatling Enterprise")
+  val enterpriseTeamId = settingKey[String]("Target team ID on Gatling Enterprise")
```
Comment (tpetillot) [inline src/main/scala/io/gatling/sbt/GatlingKeys.scala:40]: `enterpriseDefaultSimulationClassname` for coherence?

File: src/main/scala/io/gatling/sbt/settings/gatling/EnterpriseSettings.scala [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+      val defaultSimulationClassname = (config / enterpriseDefaultSimulationClassname).value
+      if (defaultSimulationClassname.isEmpty) {
+        throw new IllegalStateException(
+          s"${config.id} / enterpriseDefaultSimulationClassname has not been specified. See https://gatling.io/docs/gatling/reference/current/extensions/sbt_plugin/."
+        )
+      }
+      logger.success("Creating and starting simulation...")
+      val settingTeamId = (config / enterpriseTeamId).value;
+      val simulationAndRunSummary = client.createAndStartSimulation(
+        if (settingTeamId.isEmpty) null else UUID.fromString(settingTeamId),
```
Comment (tpetillot) [inline src/main/scala/io/gatling/sbt/settings/gatling/EnterpriseSettings.scala:139]:
```suggestion
      val optionalTeamId = Option((config / enterpriseTeamId).value).filter(_.nonEmpty).map(UUID.fromString);
      val simulationAndRunSummary = client.createAndStartSimulation(
        optionalTeamId .orNull,
```

**Pull request closed without merging**